nf_core/lint/schema_description.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+
+from logging import warn
+import nf_core.schema
+
+
+def schema_description(self):
+    """Check that every parameter in the schema has a description
+
+    The ``nextflow_schema.json`` pipeline schema should describe every flat parameter
+    Furthermore warns about parameters outside of groups
+
+    * Warning: Parameters in ``nextflow_schema.json`` without a description
+    * Warning: Parameters in ``nextflow_schema.json`` that are defined outside of a group
+    """
+    passed = []
+    warned = []
+    failed = []
+    ignored = []
+
+    # First, get the top-level config options for the pipeline
+    # Schema object already created in the `schema_lint` test
+    self.schema_obj = nf_core.schema.PipelineSchema()
+    self.schema_obj.get_schema_path(self.wf_path)
+    self.schema_obj.get_wf_params()
+    self.schema_obj.no_prompts = True
+    self.schema_obj.load_lint_schema()
+
+    # Get parameters that should be ignored according to the linting config
+    ignore_params = self.lint_config.get("schema_description", [])
+
+    # Get ungrouped params
+    if "properties" in self.schema_obj.schema.keys():
+        ungrouped_params = self.schema_obj.schema["properties"].keys()
+        for up in ungrouped_params:
+            if up in ignore_params:
+                continue
+            warned.append(f"Ungrouped param in schema: `{up}`")
```

Comment:
Almost - we need to report the tests that were ignored though..

```suggestion
            if up in ignore_params:
                ignored.append(f"Ignored ungrouped param in schema: `{up}`")
            else:
                warned.append(f"Ungrouped param in schema: `{up}`")
```